Update internal-telemetry.md (#6799)
open-telemetry/opentelemetry.io: content/en/docs/collector/internal-telemetry.md

A more detailed list is available in the following sections.

### Metrics naming

Metrics names may be altered slightly with a suffix appended or by replacing a dot (`.`) with an underscore (`_`) .

#### Suffix appended

By default and unique to Prometheus, the Prometheus exporter adds a `_total` suffix to some metrics to follow Prometheus naming conventions. For example,  `otelcol_exporter_send_failed_spans_total`. 
However in the list of internal metrics within this document, this is listed as `otelcol_exporter_send_failed_spans`. 

This behavior can be disabled by setting without_type_suffix: false in the exporter's configuration. Any internal metrics exported through OTLP will not have this behavior.
If you leave out service::telemetry::metrics::readers in the Collector config, the default Prometheus exporter set up by the Collector already has that option set. However, if you customize the readers and add a Prometheus exporter manually, you must set that option to return to the "raw" metric name.

#### Dot replaced with underscore

> `http_` and `rpc_` metrics come from instrumentation libraries. While the original metric name uses dots (`.`), when
> exposing internal metrics with Prometheus, the dots change to underscores (`_`) to match Prometheus' naming constraints.

Reviewer: I think based on @songy23's insight, we should add a versioning note to this naming convention. 

@songy23, do we need to make any changes to the full metric names in the List of Metrics? I don't see any dots there. If changes are necessary, we can do that in a follow-up PR.
 
```suggestion
`http_` and `rpc_` metrics come from instrumentation libraries. Prior to Collector v0.120.0, internal metrics exposed with Prometheus changed dots (`.`) to underscores (`_`) to match Prometheus naming constraints. Versions 0.120.0 and later of the Collector use Prometheus 3.0 scrapers, so the original 'http_` and `rpc_` metric names with dots are preserved. For more information, see the [Collector v0.120.0 release notes](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CHANGELOG.md#v01200).
```

> 

Author: @tiffany76 Did you add the note already? Pls let me know.

> 

Reviewer: Hi @BeverlyJaneJ, I did not add it. The quote in my previous comment is from the existing admonition on the page. You can use it as a basis to work the note into the section. Thanks!

A more detailed list is available in the following sections.

### Metrics naming

Metrics names may be altered slightly with a suffix appended or by replacing a dot (`.`) with an underscore (`_`) .

> ####Suffix appended

Reviewer 2: ```suggestion
#### Suffix appended
```

By default and unique to Prometheus, the Prometheus exporter adds a `_total` suffix to some metrics to follow Prometheus naming conventions. For example,  `otelcol_exporter_send_failed_spans_total`. 
However in the list of internal metrics within this document, this is listed as `otelcol_exporter_send_failed_spans`. 

This behavior can be disabled by setting without_type_suffix: false in the exporter's configuration. Any internal metrics exported through OTLP will not have this behavior.
If you leave out service::telemetry::metrics::readers in the Collector config, the default Prometheus exporter set up by the Collector already has that option set. However, if you customize the readers and add a Prometheus exporter manually, you must set that option to return to the "raw" metric name.

For more information, refer to [Collector v1.25.0/v0.119.0 release notes](https://github.com/codeboten/opentelemetry-collector/blob/313167505b44e5dc9a29c0b9242cc4547db11ec3/CHANGELOG.md#v1250v01190).

> ####Dot replaced with underscore

Reviewer 2: ```suggestion
#### Dot replaced with underscore
```

`http_` and `rpc_` metrics come from instrumentation libraries. While the original metric name uses dots (`.`), when

> exposing internal metrics with Prometheus, the dots change to underscores (`_`) to match Prometheus' naming constraints.

Reviewer 2: This is no longer the case in prometheus 3.0, see http://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CHANGELOG.md#-breaking-changes--5

Author: Removed. Thanks!

has `without_type_suffix` set to `false`. However, if you customize the readers
and add a Prometheus exporter manually, you must set that option to return to
the "raw" metric name. For more information, see the
[Collector v1.25.0/v0.119.0 release notes](https://github.com/codeboten/opentelemetry-collector/blob/313167505b44e5dc9a29c0b9242cc4547db11ec3/CHANGELOG.md#v1250v01190).

Internal metrics exported through OTLP do not have this behavior. The
[internal metrics](#lists-of-internal-metrics) on this page are listed in OTLP
format, such as `otelcol_exporter_send_failed_spans`.

#### Dots (`.`) v. underscores (`_`)

`http*` and `rpc*` metrics come from instrumentation libraries. Their original
names used dots (`.`). Prior to Collector v0.120.0, internal metrics exposed
with Prometheus changed dots (`.`) to underscores (`_`) to match Prometheus
naming conventions, resulting in metric names that looked like
`rpc_server_duration`.

Versions 0.120.0 and later of the Collector use Prometheus 3.0 scrapers, so the
original `http*` and `rpc*` metric names with dots are preserved. The
[internal metrics](#lists-of-internal-metrics) on this page are listed in their

> original form, such as`rpc.server.duration`. For more information, see the

Reviewer: I've preemptively added this statement, but I can also remove it. @songy23 or @jade-guiton-dd, do you have any thoughts on whether we should change the `http*` and `rpc*` metrics to have dots in the list of metrics below?

Reviewer 3: I think it would make sense to change them to dots. This should be less confusing for users exporting internal metrics through OTLP, and hopefully Prometheus exporter users will read through the new section.

Reviewer 3: Note: the underscores in `requests_per_rpc`, `responses_per_rpc`, `active_requests`, `open_connections` are part of the original metric names and shouldn't be replaced by dots ([reference 1](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/metrics/semantic_conventions/http-metrics.md), [reference 2](https://github.com/open-telemetry/semantic-conventions/blob/v1.34.0/docs/http/http-metrics.md)).

While checking this, I noticed that the list of `http_` metrics is already completely outdated since I updated it 7 months ago... I'll file a separate PR later to update them

Reviewer: Thanks, @jade-guiton-dd! Sorry for the delay. I was away for a few days and am still catching up. I'll make the changes to this PR this week so we can get your follow-up PR merged as well.
